src/pages/Trade/components/OrderBookCard/MyTrades.tsx

```diff
@@ -1,49 +1,110 @@
+import { CurrencyKey } from 'constants/currency';
+
+const MAX_TRADES = 100;
 
 const mapStateToProps = (state: RootState) => ({
-	trades: getMyTrades(state),
-	isLoading: getIsLoadingMyTrades(state),
-	isLoaded: getIsLoadedMyTrades(state),
+	walletAddress: getCurrentWalletAddress(state),
+	isWalletConnected: getIsWalletConnected(state),
 });
 
-const mapDispatchToProps = {
-	fetchMyTradesRequest,
-};
-
-const connector = connect(mapStateToProps, mapDispatchToProps);
+const connector = connect(mapStateToProps);
 
 type PropsFromRedux = ConnectedProps<typeof connector>;
 
 type MyTradesProps = PropsFromRedux;
 
-const MyTrades: FC<MyTradesProps> = ({ fetchMyTradesRequest, trades, isLoading, isLoaded }) => {
-	useEffect(() => {
-		fetchMyTradesRequest();
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, []);
+type SettledTrade = {
+	exchangeTimestamp: number;
+	dest: CurrencyKey;
+	src: CurrencyKey;
+	amount: number;
+	rebate: number;
+	reclaim: number;
+};
+type SettledTrades = SettledTrade[];
+
+const MyTrades: FC<MyTradesProps> = ({ walletAddress, isWalletConnected }) => {
+	const settledTradesQuery = useQuery<SettledTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(walletAddress || ''),
+		() =>
+			snxData.exchanger.exchangeEntriesSettled({
+				from: walletAddress,
+				max: MAX_TRADES,
+			}),
+		{
+			refetchInterval: REFRESH_INTERVAL,
+		}
+	);
+
+	const tradesQuery = useQuery<HistoricalTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(walletAddress || ''),
+		() =>
+			snxData.exchanges.since({
+				fromAddress: walletAddress,
+				maxBlock: Number.MAX_SAFE_INTEGER,
+				max: MAX_TRADES,
+			}),
+		{
+			refetchInterval: REFRESH_INTERVAL,
+			enabled: !!settledTradesQuery.data,
+		}
+	);
+
+	let trades = normalizeTrades(tradesQuery.data || []);
+	trades = trades.map((trade) => {
+		const settledTrade =
+			!!settledTradesQuery &&
+			!!settledTradesQuery.data &&
+			settledTradesQuery.data.find(
+				(settledTrade) =>
+					trade.timestamp === settledTrade.exchangeTimestamp &&
+					settledTrade.dest === trade.toCurrencyKey &&
+					settledTrade.src === trade.fromCurrencyKey &&
+					trade.fromAmount === settledTrade.amount
+			);
+
+		trade.isSettled = false;
+		if (!!settledTrade) {
+			console.log(settledTrade);
+			trade.rebate = settledTrade.rebate;
+			trade.reclaim = settledTrade.reclaim;
+
+			// special case for when the currency is priced in sUSD
+			const feeReclaimRebateAmount =
+				trade.toCurrencyKey === SYNTHS_MAP.sUSD
+					? settledTrade.rebate - settledTrade.reclaim
+					: settledTrade.reclaim - settledTrade.rebate;
 
-	useInterval(() => {
-		fetchMyTradesRequest();
-	}, REFRESH_INTERVAL);
+			// ( shiftAmount / amount ) * price -> gets us the price shift
+			// to get the new price, we just add the price shift (which might be a negative or positive number)
+			trade.settledPrice = (feeReclaimRebateAmount / trade.toAmount) * trade.price + trade.price;
```

Comment:
The reclaim functionality still isn't working because I get different data from `snxData.exchanger.exchangeEntriesSettled` in my PR than I get from master.

Reply:
Is there part of the duck I didn't migrate over? I feel like I got everything